### PR TITLE
DEX-1156 Cancel active orders before execution in checker

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala [hunk 1]
```diff
@@ -165,6 +165,23 @@ class Checker(superConnector: SuperConnector) {
...
+    def checkEmptyListOfActiveOrders(): ErrorOr[Boolean] =
+      for {
+        _ <- dexRest.getActiveOrdersByPair(env.matcherKeyPair, assetPairInfo.assetPair)
+          .map {
+            case left: Seq[JsValue] => left.foreach { e =>
+                val id = (e \ "id").as[ByteStr]
+                dexRest.cancelOrder(id, assetPairInfo.assetPair, env.matcherKeyPair)
+                dexRest.waitForOrderStatus(id, assetPairInfo.assetPair, OrderStatus.Cancelled.name)
+              }
+            case _ => _
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala:177]: This shouldn't compile :)

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala [hunk 2]
```diff
@@ -165,6 +165,23 @@ class Checker(superConnector: SuperConnector) {
     val submitted = mkMatcherOrder(assetPairInfo.assetPair, SELL)
     val submittedId = submitted.id()
 
+    def checkEmptyListOfActiveOrders(): ErrorOr[Boolean] =
+      for {
+        _ <- dexRest.getActiveOrdersByPair(env.matcherKeyPair, assetPairInfo.assetPair)
+          .map {
+            case left: Seq[JsValue] => left.foreach { e =>
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala:172]:
```scala
def getActiveOrdersByPair(keyPair: KeyPair, assetPair: AssetPair): ErrorOr[Seq[JsValue]]
type ErrorOr[A] = Either[String, A]
```

So you'll receive either `Left`, or `Right`, but not just `Seq`

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala [hunk 3]
```diff
@@ -165,6 +165,23 @@ class Checker(superConnector: SuperConnector) {
...
+                val id = (e \ "id").as[ByteStr]
+                dexRest.cancelOrder(id, assetPairInfo.assetPair, env.matcherKeyPair)
+                dexRest.waitForOrderStatus(id, assetPairInfo.assetPair, OrderStatus.Cancelled.name)
+              }
+            case _ => _
+          }
+
+        ids <- dexRest
+          .getActiveOrdersByPair(env.matcherKeyPair, assetPairInfo.assetPair)
+          .ensure("Count of active orders > 0")(_.isEmpty)
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala:182]: This check could fail, if Matcher haven't yet processed cancels. We should check this in the loop with:
1. E.g. 5 attempts
2. A delay of 1 second between attempts

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala (hunk 1 above)
Comment (ngadiyak) [reply dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala]: fixed

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala (hunk 2 above)
Comment (ngadiyak) [reply dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala]: fixed

File: dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala (hunk 3 above)
Comment (ngadiyak) [reply dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala]: i removed retry, since we wait Cancel status there is no need to retry

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**